images/provisionkmskeys_policy_attachment.tf

```diff
@@ -0,0 +1,9 @@
+# ------------------------------------------------------------------------------
+# Attach to the ProvisionAccount role the IAM policy that allows
+# administering KMS keys in the Images account.
+# ------------------------------------------------------------------------------
+
+resource "aws_iam_role_policy_attachment" "provisionadministerkmskeys_policy_attachment" {
```

Comment:
Why is this `provisionadministerkmskeys_policy_attachment` and not simply `administerkmskeys_policy_attachment`?

Reply:
Since we use the `ProvisionAccount` role to create the KMS key for AMIs, it also needs to be able to modify and delete that key (e.g. when we `terraform apply` changes to the key or `terraform destroy` the workspace).  I don't know if a slightly less-powerful policy/role could be used for this purpose- I started with the default set of KMS admin permissions.  I can do some experimentation and see.

Reply:
@jsf9k Please take a look at https://github.com/cisagov/cool-accounts/pull/17/commits/da5fcb9e77398722ca2dd6f7310618503bce1b04 and https://github.com/cisagov/cool-accounts/pull/17/commits/22ca97cd1f7b4f3810017e8a34be891742bc6f4c and please let me know if that looks better.  I dialed back the permissions to the point where I believe the `ProvisionKMSKeys` policy is now the bare minimum required for creation/modification/deletion of KMS keys and aliases.

Reply:
The new role looks good to me.  Should we keep the AdministerKMSKeys role around and add it to our list of roles that we can assume?  Or remove it?

Reply:
The `AdministerKMSKeys` is still used as the principal (role) allowed to administer the KMS key for AMIs: https://github.com/cisagov/cool-accounts/blob/improvement/kms_key_support/images/ami_kms_key.tf#L26-L28

Reply:
Ah, OK.  I got it now. :+1: